### PR TITLE
add batch loaders to GraphenePipeline

### PR DESCRIPTION
**Pull request by prha**
This should enable batching of runs at the repository level, making use of the new `RunBatchLimit` argument to `get_runs` / `get_run_records`.

## Test Plan
BK

### CODE REVIEW

**Comment by alangenfeld** [comment]
you get any measurements on this?

**Comment by prha** [comment]
Results are promising...

Before:
Screen Shot 2022-01-14 at 10 24 01 AM

After:
Screen Shot 2022-01-14 at 10 24 07 AM

File: python_modules/dagster-graphql/dagster_graphql/implementation/loader.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    SENSOR_STATES = "sensor_states"
+
+
+class RepositoryScopedBatchLoader:
+    def __init__(self, instance, external_repository):
+        self._instance = check.inst_param(instance, "instance", DagsterInstance)
+        self._repository = check.inst_param(
+            external_repository, "external_repository", ExternalRepository
+        )
+        self._data = {}
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/implementation/loader.py:25]: typing for this for example i think would be really useful for understanding this beast

**Review by alangenfeld — CHANGES_REQUESTED** [review]
I'm on board with this - the perf wins are worth the complexity. 

Back to you to polish this up just a bit with typing, comments, and ideally some tests. Maybe some regression protection to verify that certain graphql queries only result in N instance calls via `traced`

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_sensors.py [hunk 2]
```diff
@@ -355,3 +377,22 @@ def test_sensor_tick_range(graphql_context):
...
+    )
+    assert result.data
+    assert "repositoryOrError" in result.data
+    assert "sensors" in result.data["repositoryOrError"]
+    counter = traced_counter.get()
+    counts = counter.counts()
+    assert counts
+    assert len(counts) == 2
+    assert counts.get("DagsterInstance.get_runs") == 1
+    assert counts.get("DagsterInstance.all_stored_job_state") == 1
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_sensors.py:398]: nit: clarifying comments on these tests maybe helpful

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**